server/lib/settings/migrator.ts

```diff
@@ -36,6 +59,10 @@ export const runMigrations = async (
         // something went wrong while saving file
         throw new Error('Unable to save settings after migration.');
       }
+    } else if (oldBackup) {
+      // no migration occured
+      // we save the old backup (to avoid settings.json and settings.old.json being the same)
+      fs.writeFileSync(BACKUP_PATH, oldBackup.toString());
```

Comment:
Once again, `fs.promises`.
```suggestion
await fs.promises.writeFile(SETTINGS_PATH, JSON.stringify(migrated, null, 2));
```

Would it be better to parse this json with type safety?
```ts
const fileSaved = JSON.parse(await fs.promises.readFile(SETTINGS_PATH, 'utf-8')) as AllSettings;
```

Reply:
We could but I don't really see the point of this since this variable is only used once to be compared after being stringified.